### PR TITLE
Remove cucumber and tests

### PR DESCRIPTION
**Pull request by tijmenb**
##  Context

We introduced cucumber at the start of the build to increase our understanding and communicate how things should work. Over time we've built more acceptance tests and built a lot of documentation. The cucumber tests aren't actively maintained anymore, and the tests are covered by other specs.

## Changes proposed in this pull request

This removes the tests and the cuke infrastructure itself.

## Guidance to review

Is there anything tested by cucumber that isn't tested by the specs?

## Link to Trello card

https://trello.com/c/deEWrAsx/305-we-have-cucumber-tests-but-were-not-really-maintaining-or-using-them

## Things to check

- [X] This code doesn't rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: azure/pipelines/build.yml [hunk 1]
```diff
@@ -290,13 +229,6 @@ stages:
         testRunner: JUnit
         testResultsFiles: 'testArtifacts/**/TEST-*.xml'
```
Comment (vigneshmsft) [inline azure/pipelines/build.yml:230]: this task can be removed as well

**Comment by fofr** [comment]
Where else can the business logic for these features be found and interpreted this clearly?

**Comment by tijmenb** [comment]
@fofr I think a combination of system tests, unit tests, code, design history, [in-app documentation](https://qa.apply-for-teacher-training.education.gov.uk/support/candidate-flow) and the application itself. Anything in particular you are thinking about?

**Comment by benilovj** [comment]
> Where else can the business logic for these features be found and interpreted this clearly?

The only one I'd be worried about is features/reject_by_default.feature – there's a few edge-cases in there that may need to be added ?

**Comment by duncanjbrown** [comment]
Don't want this to fall by the wayside — having a look at `reject_by_default.feature`

**Comment by duncanjbrown** [comment]
@benilovj does 212f6a6 do the job?

**Comment by benilovj** [comment]
> @benilovj does [212f6a6](https://github.com/DFE-Digital/apply-for-teacher-training/commit/212f6a6fdd2cce85bd2888bce06ae939cecb4a66) do the job?

Yep, looks good 👍

**Pull request merged**